### PR TITLE
🛠 Fix `script is loaded` condition in effect commands

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

This PR fixes `script is loaded` condition in effect commands

---
**Target Minecraft Versions:** All
**Requirements:** None
**Related Issues:**  #4467

### CODE REVIEW

**Review by APickledWalrus — COMMENTED** [review]
Maybe we should error in init if they try to use this expression in an effect command

**Comment by AyhamAl-Ali** [comment]
> Maybe we should error in init if they try to use this expression in an effect command

Changes made ✔️ but I also left the other checks in case of other scenarios.

**Comment by TPGamesNL** [comment]
Only the variant without the string should error in init

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java [hunk 1]
```diff
@@ -52,20 +54,28 @@
...
 	private File currentScriptFile;
 	
-	@SuppressWarnings({"unchecked"})
 	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+	@SuppressWarnings({"unchecked"})
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		scripts = (Expression<String>) exprs[0];
 		setNegated(matchedPattern == 1);
 		assert getParser().getCurrentScript() != null;
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java:62]: Remove this assert

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java [hunk 2]
```diff
@@ -52,20 +54,28 @@
...
-		currentScriptFile = getParser().getCurrentScript().getFile();
+		if (getParser().isCurrentEvent(EffectCommandEvent.class) && scripts == null) {
+			Skript.error("The condition 'script loaded' requires a script name argument when used in effect commands");
+			return false;
+		}
+		Config cs = getParser().getCurrentScript();
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java:67]:
```suggestion
		Config cs = getParser().getCurrentScript();
		if (cs == null && scripts == null) {
			Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
			return false;
		}
```

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java [hunk 3]
```diff
@@ -52,20 +54,28 @@
...
 		return true;
 	}
 	
 	@Override
 	public boolean check(Event e) {
 		Expression<String> scripts = this.scripts;
 		if (scripts == null) {
+			if (currentScriptFile == null)
+				return isNegated();
 			return ScriptLoader.getLoadedFiles().contains(currentScriptFile);
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java:79]: This doesn't take isNegated into account, can you fix that too

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java [hunk 4]
```diff
@@ -59,23 +59,19 @@
...
-			Skript.error("The condition 'script loaded' requires a script name argument when used in effect commands");
+		Config cs = getParser().getCurrentScript();
+		if (cs == null && scripts == null) {
+			Skript.error("The condition 'script loaded' requires a script name argument when used outside of script files");
 			return false;
 		}
-		Config cs = getParser().getCurrentScript();
-		if (cs != null)
-			currentScriptFile = cs.getFile();
+		currentScriptFile = cs.getFile();
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java:67]: Throws NPE if `cs == null`

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java [hunk 5]
```diff
@@ -59,23 +59,19 @@
...
 		return true;
 	}
 	
 	@Override
 	public boolean check(Event e) {
 		Expression<String> scripts = this.scripts;
 		if (scripts == null) {
-			if (currentScriptFile == null)
-				return isNegated();
 			return ScriptLoader.getLoadedFiles().contains(currentScriptFile);
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java]: IMO it's better not to check if it contains null, rather skip the check like you did before

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java (hunk 5 above)
Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/conditions/CondScriptLoaded.java]: Sure \:)

**Review by TPGamesNL — CHANGES_REQUESTED** [review]
Last little thing :)

**Review by TPGamesNL — APPROVED** [review]

**Pull request merged**